shared/test/unit/mocks/mock_service.js

```diff
@@ -28,8 +48,52 @@ var MockService = {
   },
   query: function(state) {
     switch (state) {
+      case 'isUpgrade':
```

Comment:
The number of things under Service is growing at quite a fast speed. Is this the long term plan and is this scalable? It seems like just another namespace besides window to hold global soup.